torch/distributed/fsdp/_runtime_utils.py

```diff
@@ -1061,11 +1072,13 @@ def _finalize_params(
         flat_param = handle.flat_param
         if flat_param.requires_grad:
             if hasattr(flat_param, "_post_backward_hook_state"):
+                post_backward_hook_state_len = len(flat_param._post_backward_hook_state)
                 _p_assert(
-                    len(flat_param._post_backward_hook_state) == 2,
+                    post_backward_hook_state_len == 1
```

Comment:
why can it be 1 or 2 now?

Reply:
When the parameters are frozen (`requires_grad=False`), we do not have an `AccumulateGrad` object anymore, so the state looks like:
```
handle.flat_param._post_backward_hook_state = hook_handle
```

Normally, it is both the `hook_handle` and the `acc_grad`.